### PR TITLE
Migrate to Confluent Kafka proxy REST API v2

### PR DESCRIPTION
**Pull request by karsov**
UPPSF-970 Update the message-queue-consumer library to use the v2 of the REST proxy API

The implementation in 
src/main/java/com/ft/message/consumer/health/CanConnectToMessageQueueProxyHealthcheck.java
has to be evaluated if it is used and to be migrated also.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,10 +1,53 @@
 version: 2
 jobs:
   build:
+    working_directory: ~/resilient-jersey-wrapper
```
Comment (gkazakov111) [inline .circleci/config.yml]: why in the beginning working_direktory is resilient-jersey-wrapper and when publishing it is message-queue-consumer ? Please correct it.

File: src/main/java/com/ft/message/consumer/config/MessageQueueConsumerConfiguration.java [hunk 2]
```diff
@@ -28,7 +28,8 @@ public MessageQueueConsumerConfiguration(@JsonProperty("topicName") String topic
         this.autoCommit = autoCommit;
         this.backoffPeriod = backoffPeriod == 0? 8000 : backoffPeriod;
         this.streamCount = streamCount == 0? 1 : streamCount;
-        this.offsetReset = offsetReset == null? "largest" : offsetReset;
+        // Any value of offsetReset other than "latest", "earliest" or "none" causes a Kafka consumer exception
+        this.offsetReset = offsetReset == null? "latest" : offsetReset;
```
Comment (gkazakov111) [inline src/main/java/com/ft/message/consumer/config/MessageQueueConsumerConfiguration.java]: Please use here StringUtils.isBlank with static import - it is much more safe and checks for string "         " also, not only null strings.

File: src/main/java/com/ft/message/consumer/proxy/MessageQueueProxyServiceImpl.java [hunk 3]
```diff
@@ -86,6 +90,62 @@ public void destroyConsumerInstance(URI consumerInstance) {
...
+    public void subscribeConsumerInstanceToTopic(URI consumerInstance) {
+        ClientResponse clientResponse = null;
+        try {
+            URI uri = UriBuilder.fromUri(consumerInstance)
+                    .path("subscription")
+                    .build();
+
+            WebResource.Builder builder = proxyClient.resource(uri).getRequestBuilder();
+            builder.header("Content-Type", KAFKA_PROXY_V2);
+            if (queueIsNotEmpty()) {
```
Comment (karsov) [inline src/main/java/com/ft/message/consumer/proxy/MessageQueueProxyServiceImpl.java]: Can we use the same logic as in other methods here (with addProxyPortAndHostInUri(uriBuilder)) and revert the changes for rebuilding the URI in "createConsumerInstance".

File: src/main/java/com/ft/message/consumer/health/CanConnectToMessageQueueProxyHealthcheck.java [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 import com.ft.message.consumer.config.HealthcheckConfiguration;
 import com.ft.message.consumer.config.MessageQueueConsumerConfiguration;
-import com.ft.message.consumer.proxy.model.CreateConsumerInstanceResponse;
+import com.ft.message.consumer.proxy.model.ConsumerInstanceResponse;
```
[the remark this answers is not in the document]
Comment (mchompalova) [reply src/main/java/com/ft/message/consumer/health/CanConnectToMessageQueueProxyHealthcheck.java]: Because it's not a good idea to have a verb in the class name :)

File: src/main/java/com/ft/message/consumer/config/MessageQueueConsumerConfiguration.java (hunk 2 above)
Comment (mchompalova) [reply src/main/java/com/ft/message/consumer/config/MessageQueueConsumerConfiguration.java]: I used Strings.isNullOrEmpty()

Comment (gkazakov111) [reply src/main/java/com/ft/message/consumer/config/MessageQueueConsumerConfiguration.java]: Ok

**Comment by karsov** [comment]
Looks good, let's merge it.
